libs/wingsdk/src/util/child-process.ts

```diff
@@ -0,0 +1,146 @@
+import { spawn } from "child_process";
+import { Output, SpawnOptions, Stdio } from "./util";
+import { InflightClient } from "../core";
+
+/**
+ * Handle to a running child process.
+ */
+export class ChildProcess {
+  /**
+   * @internal
+   */
+  public static _toInflightType(): string {
+    return InflightClient.forType(__filename, this.name);
+  }
+
+  /**
+   * The underlying native child process object representing the spawned process.
+   */
+  private readonly child: ReturnType<typeof spawn>;
+  /**
+   * The child's OS-assigned process ID.
+   */
+  public readonly pid: number | undefined;
+  /**
+   * The accumulated standard output from the child process.
+   */
+  private stdout: string = "";
+  /**
+   * The accumulated standard error from the child process.
+   */
+  private stderr: string = "";
+  /**
+   * The exit status of the child process. Null if the process has not yet finished.
+   */
+  private exitStatus: number | null = null;
+
+  constructor(program: string, args: string[], opts?: SpawnOptions) {
+    const spawnOpts = {
+      cwd: opts?.cwd,
+      env: opts?.inheritEnv
+        ? { ...process.env, ...opts.env }
+        : { ...opts?.env },
+      stdio: [
+        opts?.stdin ?? Stdio.PIPED,
+        opts?.stdout ?? Stdio.PIPED,
+        opts?.stderr ?? Stdio.PIPED,
+      ],
+    };
+
+    // Spawn the child process with the provided options
+    this.child = spawn(program, args, spawnOpts);
+
+    this.pid = this.child.pid;
+    console.log(`Spawned child process with PID: ${this.pid}`);
+
+    // If stdio for stdout is set to `pipe`, listen for data events on stdout.
+    // if (this.child.stdout) {
+    //   this.child.stdout.on("data", (data: Buffer) => {
+    //     console.log("Received stdout data:", data.toString());
+    //     this.stdout += data.toString();
+    //   });
+    // }
+
+    // // If stdio for stderr is set to `pipe`, listen for data events on stderr.
+    // if (this.child.stderr) {
+    //   this.child.stderr.on("data", (data: Buffer) => {
+    //     console.log("Received stderr data:", data.toString());
+    //     this.stderr += data.toString();
+    //   });
+    // }
```

Comment:
why are those commented?

Reply:
Adding those causes the `stdout` to be duplicated somehow.